src/main/java/hudson/remoting/forward/CopyThread.java

```diff
@@ -20,6 +20,9 @@ public CopyThread(String threadName, InputStream in, OutputStream out) {
         super(threadName);
         this.in = in;
         this.out = out;
+        setUncaughtExceptionHandler((t, e) -> {
+            LOGGER.log(Level.WARNING, "Uncaught exception in CopyThread " + t, e);
+        });
```

Comment:
Thinking about this - how would that restart work?  The `finally` block in the previous `run` would close the input and output streams, meaning that by the time it tries again, it can't use those streams.  Perhaps the input and output stream closures should be moved elsewhere?  Or is there a simpler solution that I'm not thinking of?